### ISSUE TITLE
Updating store/signal inside effect

### ISSUE THREAD

**Issue by clinuxrulz** [solidjs/solid#588]
Here is something that might break automatic dependence detection.

```
let [ x, setX ] = createSignal(3);
let [ a, setA ] = createSignal(1);
let b = createMemo(() => x() - 1);
let [ c, setC ] = createSignal(0);
createEffect(() => setC(b() * 2));
let d = createMemo(() => a() + c());
```

In the above example I do not think node "d" will know that "b" needs to be updated before itself in a transaction. Is setting the state of Signals/Stores allowed from inside an Effect at the moment? If not, then maybe log a warning to the console?

It might look strange from the example above. But here is one use case of setting a Store from inside an Effect:

https://gist.github.com/clinuxrulz/95e6ce59e08277610d0d07bc5f8b0314

Basically I need events from the parent component to flow down to the child incase the user drags the window too fast and looses grip of the title bar.

I don't the backend implement well. Does node "d" know that node "b" needs to be updated first somehow? If not should we have a ```post(() => ...)``` function for executing stuff after the current transaction ends?

**Comment by clinuxrulz** [comment]
Had a read of Signal.ts, and looks like execution of effects is already differed until end of transaction. So should be no surprise missing updates of nodes. Sorry I should of peeked at the code first.

**Comment by ryansolid** [comment]
Yeah any new set is going to not be able to be smart enough to detect since the computation doing the setting isn't a dependency so we can't anticipate that cahnge. You are better off deriving values where possible. Effects run after computations anyway so it will need to re-calculate anyway. You should view each new set here as starting over.
